app/assets/javascripts/discourse/lib/ajax.js

```diff
@@ -139,6 +139,14 @@ export function ajax() {
       });
     };
 
+    if (args.method) {
+      args.type = args.method;
+      delete args.method;
+
+      // eslint-disable-next-line no-console
+      console.warn("`method` option is deprecated, use `type` instead");
```

Comment:
Can we use the `deprecated` helper here for consistency?

https://github.com/discourse/discourse/blob/7ec124fc897dced11eaf2e658011cb09b99f25e7/app/assets/javascripts/discourse-common/lib/deprecated.js.es6#L1

Reply:
And also some short reason for the deprecation would be useful